ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -394,17 +403,19 @@ public void onNext(InstallSnapshotReplyProto reply) {
     @Override
     public void onError(Throwable t) {
       if (!isAppenderRunning()) {
-        LOG.info("{} is stopped", this);
+        LOG.info("{} is stopped", GrpcLogAppender.this);
         return;
       }
-      LOG.error("{}: Failed installSnapshot: {}", this, t);
+      GrpcUtil.warn(LOG, () -> this + ": Failed InstallSnapshot", t);
+      grpcServerMetrics.onRequestRetry(); // Update try counter
       resetClient(null);
-      close();
     }
 
     @Override
     public void onCompleted() {
-      LOG.info("{}: follower responses installSnapshot COMPLETED", this);
+      if (!firstResponseReceived) {
```

Comment:
Why do we need this check?

Reply:
I had added it as otherwise the Log is flooded with "InstallSnapshot COMPLETED" messages even when the install snapshot is in progress. I updated it to not print in case of install snapshot notification.